common/djangoapps/student/models.py

```diff
@@ -1589,3 +1589,11 @@ def user_can_skip_entrance_exam(cls, user, course_key):
             except EntranceExamConfiguration.DoesNotExist:
                 can_skip = False
         return can_skip
+
+
+class LanguageProficiency(models.Model):
+    """
+    Represents a user's language proficiency.
+    """
+    user_profile = models.ForeignKey(UserProfile, db_index=True, related_name='language_proficiencies')
```

Comment:
I wonder if this should be off of the User instead of UserProfile. The accounts API is supposed to hide where things live (we are just using existing structures, but would probably design them differently if starting from scratch). @jimabramson and @ormsbee, what do you think? Is there any performance implication of associating this with profile instead of user?

Also, should both of these fields be indexed?